tests/backend_api/test_connector.py

```diff
@@ -61,20 +55,22 @@ def test_request_success(
     expected_response: dict[str, Any],
     expected_kwargs: dict[str, Any],
 ) -> None:
-    mocked_send_request = MagicMock(
-        name="_send_request",
-        spec=BackendApiConnector._send_request,
-        return_value=mocked_response,
-    )
-    monkeypatch.setattr(BackendApiConnector, "_send_request", mocked_send_request)
+    mocked_session = MagicMock(spec=requests.Session, name="backend_session")
```

Comment:
this test is not really backend_api specific anymore (it's only testing logic from http-connector). could you move it to `test_http.py` or remove it if it's redundant

Reply:
same goes for test_request_error_mocked

Reply:
updated in 100b4c54e8e96cc2cb59169b31594dbc0a08646a